packages/codemirror/src/editor.ts

```diff
@@ -501,6 +501,12 @@ export class CodeMirrorEditor implements CodeEditor.IEditor {
     this._editor.scrollIntoView(pos, margin);
   }
 
+  scrollIntoViewCentered(pos: CodeMirror.Position): void {
+    const top = this._editor.charCoords(pos, 'local').top;
+    const height = this._editor.getWrapperElement().offsetHeight;
+    this._editor.scrollTo(null, top - height / 2);
```

Comment:
Would it make sense to also center on the cell first, like using:

```
    this.host.scrollIntoView({
      behavior: 'auto',
      block: 'center',
      inline: 'center'
    });
    this._editor.scrollTo(null, top - height / 2);
```
?

Currently scrolling in the debugger panel editor view works nice but the one in the notebook seems to only kick in when the cell is already out of view. I do not know if this was intended, and not sure if the proposed change would be a substantial improvement.

Reply:
This does seem to make it better. Thanks! Commit added so you can try the binder again.